.github/workflows/static-analysis.yml

```diff
@@ -110,3 +114,25 @@
       run: |
         pip install --user reuse
         ~/.local/bin/reuse lint
+  scorecard-analysis:
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed for SARIF scanning upload.
+      security-events: write
+      # Needed for GitHub OIDC token if `publish_results` is true.
+      id-token: write
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v4
```

Comment:
Hmm, we now have plenty of such "not pinned by hash" warnings, but I'm unsure whether we really want to go the inconvenient route of pinning by hash. What do you think @oss-review-toolkit/core-devs? And even if e.g. Renovate would be able to bump hashes, security people would probably argue that you cannot trust an automated tool like Renovate...

Reply:
I don't like it at all, is highly more possible to a project forget that is using a hashed version with some error ( even security flaw ) that was bumped later.
Technically add more burden to maintain a project and is counter productive.

Reply:
I guess we could just merge as-is and dismiss these alerts for now, and watch how often they pop up again in the future (I assume only on action version updates).

Reply:
Yes, Renovate supports updating hashes and even puts a comment with the actual version next to it. They are using this in their own repositories, see for example:
https://github.com/renovatebot/renovate/blob/911c572da48ea7791122a0ed53cbe5e7a2d55e87/.github/workflows/build.yml#L97

So manual maintenance would not be required. However, if those actions are updated frequently it might still be annoying.